### ISSUE TITLE
Referential integrity for useToaster

### ISSUE THREAD

**Issue by guidone** [rsuite/rsuite#3349]
### What version of rsuite are you using?

5.37.0

### What version of React are you using?

18.2.0

### What browser are you using?

Chrome

### Describe the Bug

The `useToaster` hooks should keep same reference across re-renders, the code [here](https://github.com/rsuite/rsuite/blob/b4cb5f9687cc4398f0c149a311ef2efe879f441e/src/toaster/useToaster.ts#L10)

```js
const useToaster = () => {
  const { toasters } = useCustom();

  return {
  // ...
```

creates a new instance on each render, this is voiding all `useCallback`s and `useMemo`s for which the `const toaster = useToaster();` is a dependency.
In my case it's causing a cascade effect which makes the view completely unusable.

The returned object should be wrapped in a useMemo

**Comment by SevenOutman** [comment]
I'm routing this issue to @simonguo as it's related to Toaster API.

**Comment by simonguo** [comment]
Fixed in [v5.44.0](https://github.com/rsuite/rsuite/releases/tag/v5.44.0)
